packages/block-editor/src/components/list-view/index.js

```diff
@@ -27,14 +32,24 @@ import { store as blockEditorStore } from '../../store';
 
 const noop = () => {};
 const expanded = ( state, action ) => {
+	const nextState = { ...state };
+
 	switch ( action.type ) {
-		case 'expand':
-			return { ...state, ...{ [ action.clientId ]: true } };
-		case 'collapse':
-			return { ...state, ...{ [ action.clientId ]: false } };
-		default:
-			return state;
+		case 'expand': {
+			castArray( action.clientIds ).forEach( ( clientId ) => {
+				nextState[ clientId ] = true;
```

Comment:
Not sure if this is any better, but I had an alternative [here](https://github.com/WordPress/gutenberg/pull/36098/files#diff-459a77c25699c29accc1b4d398cc430a74307493f1da31b95514ea924297a527) to just expect an array in the reducer.

While getting the root level ids from the tree like this:

```js
		setExpandedState( {
			type,
			clientIds: clientIdsTree.map( ( { clientId } ) => clientId ),
		} );
```

Reply:
Not sure about the `expandAll`/`collapseAll` actions that resets the store. They will effect all levels right? I personally prefer that only the level you expand/collapse is effected and everything else stays the same.

Since the branch assumes is expanded when the key doesn’t exists we could remove the key from the store instead of setting `false`, to keep it sparse.

Yes. Expecting an array would remove the need for `castArray`, so I agree that the reducer can done in another way 👍 

I'll take another look at it later this week.